src/IO/S3/Credentials.cpp

```diff
@@ -686,7 +689,38 @@ S3CredentialsProviderChain::S3CredentialsProviderChain(
     /// because it's manually defined by the user
     if (!credentials.IsEmpty())
     {
-        AddProvider(std::make_shared<Aws::Auth::SimpleAWSCredentialsProvider>(credentials));
+        if (credentials_configuration.role_arn.empty())
+            AddProvider(std::make_shared<Aws::Auth::SimpleAWSCredentialsProvider>(credentials));
+        else
+        {
+            auto sts_client_config = Aws::STS::STSClientConfiguration();
+
+            if (!credentials_configuration.sts_endpoint_override.empty())
+            {
+                const String & new_endpoint = Poco::toLower(credentials_configuration.sts_endpoint_override);
+                if (new_endpoint.starts_with("https"))
+                {
+                    sts_client_config.scheme = Aws::Http::Scheme::HTTPS;
+                    sts_client_config.endpointOverride = new_endpoint.substr(8);
+                }
+                else
+                {
+                    sts_client_config.scheme = Aws::Http::Scheme::HTTP;
```

Comment:
Are we sure that sts_endpoint_override is always in correct format?
`t.me/noaws.amazon.com/` is interpreted here as `http://aws.amazon.com/`.
May be better parse with `Poco::URI` to check schema?